docs/local_deployment.md

```diff
@@ -0,0 +1,121 @@
+# Local Deployment
+
+When using nested templates against Azure, they need to be publicly accessible. This is not easy to accomplish if you are developing on a laptop within a corporate network for example. To assist with this a local script has been setup that will perform the deployment for you.
+
+When the script stages are executed the following occurs:
+
+ - Does the resource group already exist
+   - If this is the first time the script has been run this will be false
+   - If this is not the first time, then a local file `.deploy` will be read and the iteration found. The resource group name is thus `<RESCOURCE_GROUP>-<ITERATION>`. Delete the resource group.
+ - Increment the iteration and write out to the `.deploy` file
+ - Determine the name of the resource group to create, again `<RESCOURCE_GROUP>-<ITERATION>`
+ - Upload files to the specified storage account and container
+ - Perform deployment into the newly created resource group
+
+## Limitations
+
+The are currently some few limitations on the tooling. These will be addressed
+
+ - The storage account and container must already exist
```

Comment:
Good spot. Currently the tool will just crash out
In this iteration I will just fail with an error message, but this will change to create a storage account if it does not exist eventually.